### PR TITLE
Javascript/javascript2/week1

### PR DESCRIPTION
**Pull request by Gohar33**

### CODE REVIEW

File: javascript/javascript2/week1/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//Find the shortest word
+
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"]
+
+function thisFunctionName(danishWords) {
```
Comment (sboutzen) [inline javascript/javascript2/week1/index.js:5]: :trollface:

Comment (sboutzen) [reply javascript/javascript2/week1/index.js:5]: If this was a code base you had to maintain over time, you should choose another function name like `getShortestWord`

Comment (sboutzen) [reply javascript/javascript2/week1/index.js:5]: In my opinion map/reduce is a cleaner way to do it than a for loop, but this is nitpicking

File: javascript/javascript2/week1/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+//Find and count the Danish letters
+
+const danishString = "Jeg har en blå bil";
+const danishString2 = "Blå grød med røde bær";
+
+function notThisFunctionName(letters) {
+    let result = { "total": 0, "æ": 0, "ø": 0, "å": 0 }
+
+    for (let i = 0; i < letters.length; i++) {
```
Comment (sboutzen) [inline javascript/javascript2/week1/index.js:26]: I would prefer map as mentioned before, but again, nitpicking

Comment (sboutzen) [reply javascript/javascript2/week1/index.js:26]: When defining a function, try to capture the essence of the function in the function name.

If you have well named functions, a lot of code can be read like prosa, which can dramatically increase the speed
with which other devs (or your future self reading old code) will understand your code. We do significantly more reading code than writing code

**Review by sboutzen — COMMENTED** [review]
Well done, small comments. 💪

File: javascript/javascript2/week1/index.js (hunk 1 above)
Comment (Gohar33) [reply javascript/javascript2/week1/index.js:5]: Hi Simon, thank you for your feedback. Map/reduce/filter we should learn in our upcoming class. However, I tried to implement it now in this exercise. I don't understand reduce yet so I tried with map and I've got such result [undefined, undefined, undefined, undefined, "ø", undefined] and  I couldn't find a way to exclude other elements. 

Then I changed it to filter and got a new array with only  "ø". This is my code:

 const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"]

let shortestWord = danishWords.filter(function (shortWord) {
    let newArray = shortWord[0]
    if (shortWord.length <= newArray.length) {
        newArray = shortWord
        return newArray;

    }

})

console.log(shortestWord);

File: javascript/javascript2/week1/index.js (hunk 2 above)
Comment (Gohar33) [reply javascript/javascript2/week1/index.js:26]: Thank you I'll go through again and will change all names :)

File: javascript/javascript2/week1/index.js (hunk 1 above)
Comment (sboutzen) [reply javascript/javascript2/week1/index.js:5]: I just realized this is next weeks curriculum. For now disregard the comment. I will say though that when you are in doubt about how any of these functions work, try and google them. Mozilla has good javascript documentation. Googling 'javascript map' yields this link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Comment (Gohar33) [reply javascript/javascript2/week1/index.js:5]: Thank you Simon :)

**Pull request merged**